CRM/Contact/BAO/Query.php

```diff
@@ -5829,12 +5811,40 @@ public function convertToPseudoNames(&$dao, $return = FALSE, $usedForAPI = FALSE
         $val = $dao->{$value['idCol']};
         if ($key == 'groups') {
           $dao->groups = $this->convertGroupIDStringToLabelString($dao, $val);
-          return;
+          continue;
```

Comment:
@eileenmcnaughton That will be problematic in PHP7 https://github.com/wimg/PHPCompatibility/issues/292